src/tests/junit/org/apache/tools/ant/taskdefs/JavacTest.java

```diff
@@ -244,4 +248,85 @@ public void testTargetOverridesDefault() {
         javac.setTarget("1.5");
         assertEquals("1.5", javac.getTarget());
     }
+
+    @Test
+    public void testModulesourcepathOrSrcDirRequired() {
+        try {
+            javac.checkParameters();
+            fail("Build exception should have been thrown - neither srcDir nor modulesourcepath");
+        } catch (BuildException e) {
+            //pass
```

Comment:
You can use `@Test(expected = BuildException.class)` instead of useless try/catch.